src/firstboot/firstboot.c

```diff
@@ -667,6 +721,31 @@ static int process_root_password(void) {
         if (lock < 0)
                 return log_error_errno(lock, "Failed to take a lock: %m");
 
+        if (arg_delete_root_password) {
+                const char *etc_passwd;
+                struct passwd *p;
+
+                if (arg_copy_root_password || arg_root_password || arg_prompt_root_password)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "--delete-root-password cannot be combined with other root password options");
```

Comment:
hmm, i figure this kind of argument checking should be done at the end of parse_argv()